vignettes/articles/tbl_ard-functions.Rmd

```diff
@@ -0,0 +1,125 @@
+
+This table has three types of data summaries: a continuous variable summary for AGE, a categorical variable summary for AGEGR1, RACE, and ETHNIC and a dichotomous variable summary for SEX.
+
+```{r}
+library(gtsummary)
+library(cards)
+
+# summarizing both continuous and categorical variables, consolidate using ard_stack call
+ard_stack(
+  data = cards::ADSL,
+  .by = ARM,
+  ard_categorical(variables = c("AGEGR1", "RACE", "SEX", "ETHNIC")),
+  ard_continuous(variables = "AGE"),
+  .attributes = TRUE,
+  .missing = TRUE,
+  .total_n = TRUE,
+  .overall = TRUE # generate ard with overall column included
+  ) |> # pass to tbl_ard_summary to generate a table.
+  tbl_ard_summary(by = ARM, overall = TRUE) # pass ard to table building function with the overall argument.
+
+```
+
+Notice this ARD first approach builds an equivalent table to one built with `tbl_summary()`
+
+```{r}
+
+cards::ADSL |>
+  select(c("AGEGR1", "RACE", "SEX", "ETHNIC", "AGE", "ARM")) |>
+  tbl_summary(by = ARM) |>
+  add_overall() 
+
+```
+The ARD for this table can be extracted using the `gather_ard()` function.
+
+### Building a disposition table
+Here, we use tbl_ard_wide_summary to build a Pooled disposition table. 
+
+```{r}
+# all variables within tbl_ard_wide_summary must be the same summary type
+ard_stack(
+  trial,
+  ard_dichotomous(variables = response),
+  ard_categorical(variables = grade),
+  .missing = TRUE,
+  .attributes = TRUE,
+  .total_n = TRUE
+) |>
+  tbl_ard_wide_summary()
+```
+
+### Building an Adverse Events table
+
+For tables that require hierarchical or nested analysis, `ard_stack_hierarchical()` is recommended for building an ARD. The companion function for rendering the table using the ARD is `tbl_ard_hierarchical()`
+
+```{r}
+# Adverse Events Rates Table
+
+ADAE_subset <- cards::ADAE |>
+  dplyr::filter(
+    AESOC %in% unique(cards::ADAE$AESOC)[1:5],
+    AETERM %in% unique(cards::ADAE$AETERM)[1:5])
+
+ard <- cards::ard_stack_hierarchical(
+  data = ADAE_subset,
+  variables = c(AESOC, AETERM),
+  by = TRTA,
+  denominator = cards::ADSL|> mutate(TRT01A = ARM),
+  id = USUBJID
+  )
+
+tbl_ard_hierarchical(
+   cards = ard,
+   variables = c(AESOC, AETERM),
+   by = TRTA
+ )
+
+```
+
+### Complex Analysis Tables
+
+The ARD to Table pipeline is most convenient when trying to consolidate multiple analysis steps into an ARD to feed only the relevant stats to the table building machinery. In the example below, we create ARDs that compute the survival probabilities of both treatment groups, and also conduct a one sample t-test on the survival estimates. The resulting ARD is filtered to only contain the relevant estimate values which can be used generate the table we want. Note that in this pipeline, all the survival analysis stats such as confidence intervals and and standard error are retained in the ARD object - enabling reuse of this object to generate other tables. 
```

Comment:
> one sample t-test on the survival estimates

I think the language describing the analysis methods can be cleaned up. I can help you with that when i return.